controllers/volume/v1alpha1/volume_controller.go

```diff
@@ -425,8 +424,8 @@ func (r *Reconciler) createInstanceStoragePVC(
 		},
 		Spec: corev1.PersistentVolumeClaimSpec{
 			StorageClassName: &claim.StorageClass,
-			Resources: corev1.ResourceRequirements{
-				Requests: map[corev1.ResourceName]resource.Quantity{
+			Resources: corev1.VolumeResourceRequirements{
```

Comment:
Out of curiosity, does it tie us with the Kubernetes version of Supervisor?  IOW, this change in VM operator can only be shipped on a Supervisor where the new PVC APIs were introduced, right?

Reply:
Also, does it matter whether CSI has been updated to consume the latest core/v1 APIs?  I worry if they are using some sort of typed clientset (?) at a specific version which might drop the resource requirements entirely.

Reply:
The change is just client-side. `corev1.VolumeResourceRequirements` is `type VolumeResourceRequirements map[corev1.ResourceName]resource.Quantity`.

Reply:
Hi @xing-yang,

Do you know if this change to the the `PersistentVolumeClaimSpec` resource requires to `VolumeResourceRequirements` is a problem for back-compat? The [1.29 release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.29.md) indicate:

> * Go API: the ResourceRequirements struct was replaced with VolumeResourceRequirements for use with volumes. (https://github.com/kubernetes/kubernetes/pull/118653, [@pohly](https://github.com/pohly))

@aruneshpa , it looks like the `Claims` field was not even added until 1.26 anyway, so I doubt this is an issue. Especially given `VolumeResourceRequirements` and [`ResourceRequirements`](https://pkg.go.dev/k8s.io/api/core/v1#ResourceRequirements) are the same on the wire with the latter containing additional info if set (the claim names).

Reply:
Okay, CSI for vSphere [does not ever set](https://github.com/search?q=repo%3Akubernetes-sigs%2Fvsphere-csi-driver+ResourceRequirements&type=code) `ResourceRequirements.Claims` either. We should be fine since it's the same wire-format for both Go types.

Reply:
@akutz Yes, it should be fine.  We've finished testing vSphere CSI driver with K8s 1.29 and didn't run into any problems.